sematic/resolvers/tests/test_log_streamer.py

```diff
@@ -35,12 +37,16 @@ def test_ingested_logs():
 
     assert len(uploads) > 1  # should be at least one timed, and the final flush
 
-    # first upload should have only contained early writes
-    assert uploads[0] == [
-        f"{remote_prefix}\n",
-        "From stdout\n",
-        "From stderr\n",
-    ]
+    # early uploads should have only contained early writes
+    assert any(
```

Comment:
Flake mode: first upload happened before `From stderr` got written to logs. This makes it so it will be ok if the second, third, Nth log is the only one with that limited portion of the log statements. I also:

- increased the time the script pauses between the "early" logs and the "later" ones. In principle 50 uploads should happen between the early and late ones, if the OS schedules the processes completely fairly.
- added a retry to the overall test. In the rare case where the OS doesn't schedule the uploader process for 50 intervals in a row, we won't fail until that has happened on three consecutive attempts at the test.

Why do we care about this test of early vs late logs? Because we want to ensure that logs are being uploaded *as the code executes* and not just once at the very end.